code/modules/vehicle/vehicle.dm

```diff
@@ -85,33 +87,34 @@
 
 
 /obj/vehicle/bullet_act(obj/item/projectile/Proj)
-	if(buckled_mob)
+	if(buckled_mobs.len)
+		var/mob/living/buckled_mob = buckled_mobs[1]
 		buckled_mob.bullet_act(Proj)
 
 //MOVEMENT
 /obj/vehicle/relaymove(mob/user, direction)
 	if(user.incapacitated())
-		unbuckle_mob()
+		unbuckle_mob(user)
 
 	if(keycheck(user))
-		if(!Process_Spacemove(direction) || world.time < next_vehicle_move || !isturf(loc))			return
+		if(!Process_Spacemove(direction) || world.time < next_vehicle_move || !isturf(loc))
+			return
+
 		next_vehicle_move = world.time + vehicle_move_delay
 
 		step(src, direction)
 
-		if(buckled_mob)
-			if(buckled_mob.loc != loc)
-				buckled_mob.buckled = null //Temporary, so Move() succeeds.
-				buckled_mob.buckled = src //Restoring
-
-			if(istype(src.loc, /turf/simulated))
-				var/turf/simulated/T = src.loc
-				if(T.wet == TURF_WET_LUBE)	//Lube! Fall off!
-					playsound(src, 'sound/misc/slip.ogg', 50, 1, -3)
-					buckled_mob.Stun(7)
-					buckled_mob.Weaken(7)
-					unbuckle_mob()
-					step(src, dir)
+		if(buckled_mobs.len)
+			for(var/m in buckled_mobs)
+				var/mob/living/buckled_mob = m
+				if(istype(src.loc, /turf/simulated))
+					var/turf/simulated/T = src.loc
+					if(T.wet == TURF_WET_LUBE)	//Lube! Fall off!
+						playsound(src, 'sound/misc/slip.ogg', 50, 1, -3)
```

Comment:
Seriously though how do i need to fix it proper..i just tried something and it sill does not look right.